addons/point_of_sale/models/pos_config.py

```diff
@@ -330,6 +330,10 @@ def _check_companies(self):
         if any(self.available_pricelist_ids.mapped(lambda pl: pl.company_id.id not in (False, self.company_id.id))):
             raise ValidationError(_("The selected pricelists must belong to no company or the company of the point of sale."))
 
+    def _check_cash_control_journal(self):
+        if not self.payment_method_ids.cash_journal_id and self.cash_control:
```

Comment:
what you should check here is 'is_cash_count' boolean -> if not self.payment_method_ids.mapped('is_cash_count')